### DISCUSSION TITLE
Tradingview lightweight Charts Code & License ?

### DISCUSSION THREAD

**Discussion by digitalml** [klinecharts/KLineChart#354] [Q&A]
There is a comment in file KLine\src\common\MouseTouchEventHandler.ts that the code comes from Tradingview Lightweight Charts. Is this the only code in KLine charts that comes from Tradingview Lightweight Charts, or is there more?

**Comment by liihuu** [comment]
KLineChart 遵循开源协议，任何来自其它开源库的文件都会在文件头说明，并且也会在NOTICE文件中说明，同时也会有包含源头库License.
